### PR TITLE
[4.1] Fixes fetchOperation awareness for repeatable fields

### PR DESCRIPTION
**Pull request by pxpm**
This is needed with PR #2728

### CODE REVIEW

**Comment by tabacitu** [comment]
Ah! I was actually thinking about this. It's something I've stumbled upon when fixing ```select_and_order``` to work inside repeatable. 

**Do you see any reason why ```repeatable``` would store the array of IDs as a string ```1,2,3``` instead of a JSON array ```[1, 2, 3]```?** I don't think so, and I think it would be cleaner to just use JSON instead of string:
- [ ] inside repeatable, for selects, when it populates ```data-select-options```;
- [x] inside ```select_and_order``` - done; assumes array, but converts string to array if necessary;
- [ ] inside ```select2_from_ajax```;
- [ ] inside ```select2_from_ajax_multiple```;
- [ ] inside the ```Fetch``` operation (or I guess we should just NOT merge this PR in this case?);

The way I see it, JSON is cleaner, and it prevents us from using a bunch of explodes and concats everywhere. I prefer json_encode and json_decode any day 😆 

**What do you think @pxpm ?**

File: src/app/Http/Controllers/Operations/FetchOperation.php [hunk 1]
```diff
@@ -60,13 +60,13 @@ private function fetch($arg)
         $config['searchable_attributes'] = $config['searchable_attributes'] ?? $model_instance->identifiableAttribute();
         $config['query'] = isset($config['query']) && is_callable($config['query']) ? $config['query']($config['model']) : $model_instance; // if a closure that has been passed as "query", use the closure - otherwise use the model
 
-        // FetchOperation is aware of an optional parameter 'keys' that will fetch you the entity/entities that match the provided keys
```
Comment (tabacitu) [inline src/app/Http/Controllers/Operations/FetchOperation.php:63]: Hmm why did you change ```key``` to ```related_keys``` here?

File: src/app/Http/Controllers/Operations/FetchOperation.php [hunk 2]
```diff
@@ -60,13 +60,13 @@ private function fetch($arg)
...
-        if (request()->has('keys')) {
-            $decoded_keys = json_decode(request()->get('keys'));
-            if (is_array($decoded_keys)) {
-                return $model_instance->whereIn($model_instance->getKeyName(), $decoded_keys)->get();
+        // FetchOperation is aware of an optional parameter 'related_keys' that will fetch you the entity/entities that match the provided keys
+        if (request()->has('related_keys')) {
+            $array_keys = explode(',', request()->get('related_keys'));
+            if (is_array($array_keys) && count($array_keys) > 1) {
+                return $model_instance->whereIn($model_instance->getKeyName(), $array_keys)->get();
```
Comment (tabacitu) [inline src/app/Http/Controllers/Operations/FetchOperation.php:67]: `$model_instance->findMany($array_keys)` would probably be shorter here.

File: src/app/Http/Controllers/Operations/FetchOperation.php [hunk 3]
```diff
@@ -60,13 +60,13 @@ private function fetch($arg)
...
             } else {
-                return $model_instance->where($model_instance->getKeyName(), $decoded_keys)->first();
+                return $model_instance->where($model_instance->getKeyName(), $array_keys)->get();
```
Comment (tabacitu) [inline src/app/Http/Controllers/Operations/FetchOperation.php:69]: `$model_instance->find($array_keys)` would probably be shorter.

**Comment by tabacitu** [comment]
PS. I've merged this into https://github.com/Laravel-Backpack/CRUD/pull/2757/files so we have both PRs in one place and can easily test them together. So far so good! Excited to merge them!

But these questions still need answers - refactoring to JSON is something we should do from day one if we decide it's better.

File: src/app/Http/Controllers/Operations/FetchOperation.php (hunk 1 above)
Comment (pxpm) [reply src/app/Http/Controllers/Operations/FetchOperation.php:63]: Just to be clear on what those keys mean. They are model related keys, so I just updated the name to reflect that. `keys` seemed very vague.

**Comment by tabacitu** [comment]
All of the above have been fixed inside https://github.com/Laravel-Backpack/CRUD/pull/2757 - but please @pxpm let me know what your thoughts are about this - maybe I made a bad decision.

I'll close this in favour of https://github.com/Laravel-Backpack/CRUD/pull/2757

File: src/app/Http/Controllers/Operations/FetchOperation.php (hunk 2 above)
Comment (pxpm) [reply src/app/Http/Controllers/Operations/FetchOperation.php:67]: The problem I see with find is that it returns `->first()` and we want a collection so we need to use `->get()`, because `first()` returns an model instance. 

You can have a look here about what I am talking:

https://github.com/laravel/framework/blob/c4f877e03c28bdea584f062e40025dd5e5be62e8/src/Illuminate/Database/Eloquent/Builder.php#L339

Apart from that findMany() would work great I think.

File: src/app/Http/Controllers/Operations/FetchOperation.php (hunk 3 above)
Comment (pxpm) [reply src/app/Http/Controllers/Operations/FetchOperation.php:69]: read my previous concern please in findMany :)

File: src/app/Http/Controllers/Operations/FetchOperation.php (hunk 1 above)
Comment (tabacitu) [reply src/app/Http/Controllers/Operations/FetchOperation.php:63]: Oh ok. 

Let's keep `keys` - I think it's clear enough. At this point (inside FetchOperation, or inside the JS) we only work with that "related item", so of course they're that model's keys. They're the keys of the only model we're working with. At least that's what I think now - I reserve the option to say "_I was wrong, you were right_" in the future 😆

File: src/app/Http/Controllers/Operations/FetchOperation.php (hunk 2 above)
Comment (tabacitu) [reply src/app/Http/Controllers/Operations/FetchOperation.php:67]: Oh OK! Got it. Best to use ```findMany()``` or keep ```where()``` then - good catch.

Comment (tabacitu) [reply src/app/Http/Controllers/Operations/FetchOperation.php:67]: Update - I left is as it was 😄 ```where``` ftw!
